lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_emm_encode_decode.cpp

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+
+extern "C" {
+#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/AttachRequest.h"
+#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/EmmInformation.h"
+#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateRequest.h"
+#include "lte/gateway/c/core/oai/tasks/nas/emm/msg/UplinkNasTransport.h"
+#include "lte/gateway/c/core/oai/common/dynamic_memory_check.h"
+#include "lte/gateway/c/core/oai/common/log.h"
+}
+
+#include "lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.h"
+
+namespace magma {
+namespace lte {
+
+#define BUFFER_LEN 200
+
+#define FILL_EMM_COMMON_MANDATORY_DEFAULTS(msg)                                \
+  do {                                                                         \
+    msg.protocoldiscriminator = EPS_SESSION_MANAGEMENT_MESSAGE;                \
+    msg.securityheadertype    = 0x0001;                                        \
+    msg.messagetype           = 33;                                            \
+  } while (0)
+
+#define ASSERT_EQ_GUTI_EPS_MOBILE_IDENTITY(orig_guti, dec_guti)                \
+  ASSERT_EQ(orig_guti.guti.mcc_digit1, dec_guti.guti.mcc_digit1);              \
+  ASSERT_EQ(orig_guti.guti.mcc_digit2, dec_guti.guti.mcc_digit2);              \
+  ASSERT_EQ(orig_guti.guti.mcc_digit3, dec_guti.guti.mcc_digit3);              \
+  ASSERT_EQ(orig_guti.guti.mnc_digit1, dec_guti.guti.mnc_digit1);              \
+  ASSERT_EQ(orig_guti.guti.mnc_digit2, dec_guti.guti.mnc_digit2);              \
+  ASSERT_EQ(orig_guti.guti.mnc_digit3, dec_guti.guti.mnc_digit3);              \
+  ASSERT_EQ(orig_guti.guti.mme_code, dec_guti.guti.mme_code);                  \
+  ASSERT_EQ(orig_guti.guti.m_tmsi, dec_guti.guti.m_tmsi);
```

Comment:
you can simply use memcmp for basic structs that does not have any pointer fields.